src/software/new_geom/ray.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include "software/geom/line.h"
+#include "software/geom/point.h"
+
+class Ray final
+{
+   public:
+    /**
+     * Creates a degenerate Ray at (0, 0)
+     */
+    inline explicit constexpr Ray() {}
+
+    /**
+     * Creates a Ray with a start point and a direction
+     *
+     * @param start the starting point of Ray
+     * @param direction the direction of Ray
+     */
+    inline explicit Ray(const Point& start, const Vector& direction)
```

Comment:
Can we also have a constructor that takes an `Angle` for the direction?

Reply:
Addressed in fbdf4f3